eng/templates/stages/retain-build.yml

```diff
@@ -0,0 +1,23 @@
+stages:
+- stage: RetainProductionBuild
+  displayName: Retain Production Build
+  dependsOn: []
+  jobs:
+  - job: RetainBuild
+    displayName: Retain Production Build
+    pool:
+      name: NetCore1ESPool-Internal-NoMSI
+      demands: ImageOverride -equals 1es-windows-2022
+
+    variables:
+    - name: AzdoOrgUri
+      value: https://dev.azure.com/dnceng
+    - name: AzdoProject
+      value: internal
+
+    steps:
+    - template: ..\..\common\templates\steps\retain-build.yml
```

Comment:
Maybe this could be just injected as a build step in the main Build stage?
Maybe there's no reason spinning up a whole agent for this?

Reply:
I guess we could do it as the first task in the stage. I think we want to retain the build whatever happens